pkg/command/ceip_participation.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package command
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/cli"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/component"
+	configlib "github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/plugin"
+)
+
+// CeipOptOutStatus and CeipOptInStatus are constants for the CEIP opt-in/out verbiage
+const (
+	CeipOptInStatus  = "Opt-in"
+	CeipOptOutStatus = "Opt-out"
+)
+
+// Note(TODO:prkalle): The below ceip-participation command(experimental) added may be removed in the next release,
+//       If we decide to fold this functionality into existing 'tanzu telemetry' plugin
+
+func newCEIPParticipationCmd() *cobra.Command {
+	var ceipParticipationCmd = &cobra.Command{
+		Use:     "ceip-participation",
+		Short:   "Manage CEIP Participation",
+		Long:    "Manage CEIP Participation",
+		Aliases: []string{"ceip"},
+		Annotations: map[string]string{
+			"group": string(plugin.SystemCmdGroup),
+		},
+	}
+	ceipParticipationCmd.SetUsageFunc(cli.SubCmdUsageFunc)
+
+	ceipParticipationCmd.AddCommand(
+		newCEIPParticipationSetCmd(),
+		newCEIPParticipationGetCmd(),
+	)
+
+	return ceipParticipationCmd
+}
+
+func newCEIPParticipationSetCmd() *cobra.Command {
+	var setCmd = &cobra.Command{
+		Use:   "set OPT_IN_BOOL",
+		Short: "Set the opt-in preference for CEIP",
+		Long:  "Set the opt-in preference for CEIP",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := configlib.SetCEIPOptIn(args[0])
```

Comment:
Could we add a validation on `args` before SetXXX()

Reply:
Updated. Thanks